### ISSUE TITLE
Zod optional not working on non object property values

### ISSUE THREAD

**Issue by pitkes22** [StefanTerdell/zod-to-json-schema#10]
I am trying to use zod's [optional](https://github.com/colinhacks/zod#optionals) `z.optional()` but it don't seem to heve any effect on basic types. 

```TypeScript
zodToJsonSchema(z.number())

// {
//     "$schema": "http://json-schema.org/draft-07/schema#",
//     "type": "number"
// }

zodToJsonSchema(z.optional(z.number()))

// {
//     "$schema": "http://json-schema.org/draft-07/schema#",
//     "type": "number"
// }
```

but it works correctly on object properties

```TypeScript
zodToJsonSchema(z.object({
    username: z.string(),
}))

// {
//     "$schema": "http://json-schema.org/draft-07/schema#",
//     "type": "object",
//     "properties": {
//         "username": {
//             "type": "string"
//         }
//     },
//     "required": [
//         "username"
//     ],
//     "additionalProperties": false
// }

zodToJsonSchema(z.object({
    username: z.string().optional(),
}))

// {
//     "$schema": "http://json-schema.org/draft-07/schema#",
//     "type": "object",
//     "properties": {
//         "username": {
//             "type": "string"
//         }
//     },
//     "additionalProperties": false
// }
```
I Would expect it to produce a union of `undefined` and given type, which is the same result as the TypeScript type that is inferred. 

```TypeScript
const zodType = z.optional(z.number())
type exampleType = z.infer<typeof zodType>
// exampleType type: number | undefined
```

Output should be probably the same as of this code: 

```TypeScript

zodToJsonSchema(z.union([z.undefined(), z.number()]))
// {
//     "$schema": "http://json-schema.org/draft-07/schema#",
//     "anyOf": [
//         {
//             "not": {}
//         },
//         {
//             "type": "number"
//         }
//     ]
// }
```

**Comment by StefanTerdell** [comment]
This will be fixed in the next version

**Comment by StefanTerdell** [comment]
Fixed in 3.14.0 released just now
